src/schematools/types.py

```diff
@@ -277,7 +290,7 @@ def build_through_table(
         if relation is None and table.is_temporal:
             relation = field.relation
         right_dataset_id, right_table_id = [
-            to_snake_case(part) for part in relation.split(":")[:2]
+            to_snake_case(part) for part in str(relation).split(":")[:2]
```

Comment:
Dit wordt dus `"None".split(":")[:2]` en geeft nog vreemdere situaties zo.

Reply:
Goed punt, beter met een cast.

Reply:
Added extra precaution, relation should never be None.